$OpenBSD: patch-startkde_cmake,v 1.2 2013/12/30 14:11:48 zhuk Exp $
Fix lnusertemp locate and make startkde work even without X started.
--- startkde.cmake.orig	Wed Aug 28 20:59:38 2013
+++ startkde.cmake	Mon Dec  9 20:11:15 2013
@@ -3,6 +3,11 @@
 #  DEFAULT KDE STARTUP SCRIPT ( @KDE_VERSION_STRING@ )
 #
 
+case X"$DISPLAY" in
+X)
+	exec /usr/X11R6/bin/startx $0;;
+esac
+
 if test "x$1" = x--failsafe; then
     KDE_FAILSAFE=1 # General failsafe flag
     KWIN_COMPOSE=N # Disable KWin's compositing
@@ -228,10 +233,10 @@ else
     export GS_LIB
 fi
 
-lnusertemp=`kde4-config --path exe --locate lnusertemp`
+lnusertemp="`kde4-config --path libexec --locate lnusertemp`"
 if test -z "$lnusertemp"; then
   # Startup error
-  echo 'startkde: ERROR: Could not locate lnusertemp in '`kde4-config --path exe` 1>&2
+  echo 'startkde: ERROR: Could not locate lnusertemp in '`kde4-config --path libexec` 1>&2
 fi
 
 # Link "tmp" "socket" and "cache" resources to directory in /tmp
